plugin/hover.py

```diff
@@ -95,14 +95,13 @@ def run_async() -> None:
             listener = wm.listener_for_view(self.view)
             if not listener:
                 return
+            self._diagnostics_by_config, covering = listener.diagnostics_touching_point_async(hover_point)
             if not only_diagnostics:
                 self.request_symbol_hover_async(listener, hover_point)
-            self._diagnostics_by_config, covering = listener.diagnostics_touching_point_async(hover_point)
-            if self._diagnostics_by_config:
-                if not only_diagnostics:
-                    actions_manager.request_with_diagnostics_async(
-                        self.view, covering, self._diagnostics_by_config,
-                        functools.partial(self.handle_code_actions, listener, hover_point))
+                actions_manager.request_for_region_async(
+                    self.view, covering, self._diagnostics_by_config,
+                    functools.partial(self.handle_code_actions, listener, hover_point))
+            else:
```

Comment:
With the previous logic we would show the hover info as soon as possible and update the popup with diagnostics as soon as those arrive. With your change I believe that we will wait with showing the popup until diagnostics arrived?

Reply:
> With the previous logic we would show the hover info as soon as possible and update the popup with diagnostics as soon as those arrive. With your change I believe that we will wait with showing the popup until diagnostics arrived?

Diagnostic response is negligible, it does not require a request server

Reply:
`actions_manager.request_for_region_async` does make a request (potentially even multiple) to the server.

Reply:
Sorry, I don t quite understand, it seems to me that this is a request for the `code action`. It s after hovering

Reply:
In this highlighted code there are two types of requests made:
 1. one `textDocument/hover` request to get symbol information for hovered point
 2. one or more `textDocument/codeAction` requests to get code actions for hovered thing

And as said in the first comment, previously the first one would be triggered and shown without waiting for the 2 to finish.

Reply:
I too don't understand this part.

Reply:
Resolved in 17d858f